### PR TITLE
Fp16 interpreter support: Convolution

### PR DESCRIPTION
**Pull request by qcolombet**
*Description*:
This PR adds fp16 support for the convolution node in the interpreter

*Testing*:
Added test cases
*Documentation*: N/A

This depends on #1660 and is related to #1329

### CODE REVIEW

File: lib/Backends/Interpreter/InterpreterNodes.cpp [hunk 1]
```diff
@@ -31,15 +31,19 @@ using namespace glow;
 //===----------------------------------------------------------------------===//
 
 // This is the floating point implementation of Convolution.
-void InterpreterFunction::fwdConvolutionInst_FloatImpl(
+template <typename ElemTy>
+typename std::enable_if<
```
Comment (nadavrot) [inline lib/Backends/Interpreter/InterpreterNodes.cpp]: Why do we need the enable_if section? If we need to protect the usage of this method, could we insert a static_assert into the body of the function to keep the declaration simple?

File: tests/unittests/OperatorTest.cpp [hunk 2]
```diff
@@ -1291,6 +1291,92 @@ TEST_P(InterpAndCPU, IntBatchedArith) {
...
+  // In this test we generate a Floating-point based convolution and an integer
+  // convolution. We pass the same values and then subtract the results. We
+  // check that the results are below some known delta.
+
+  // In this test the output of the convolution is in the range [-256 ... 256].
+  // The inputs (specified below) are in the range [-1 .. 1],
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+
+  auto *input = mod.createVariable(ElemKind::FloatTy, {1, 10, 10, 3}, "in");
```
Comment (nadavrot) [inline tests/unittests/OperatorTest.cpp]: Please use the new createPlaceholder API.

File: tests/unittests/graphTest.cpp [hunk 3]
```diff
@@ -64,6 +65,81 @@ TEST(Graph, simpleTestConv) {
   EXPECT_GT(M.getInstrs().size(), 0);
 }
 
+/// Check that we can create convolution with float16.
+TEST(Graph, float16Conv) {
+  Module MD;
+  Function *F = MD.createFunction("F");
+  Node *K = MD.createVariable(ElemKind::Float16Ty, {4, 320, 200, 3}, "input");
```
Comment (nadavrot) [inline tests/unittests/graphTest.cpp]: ditto. Please don't add new createVariable call sites.

File: .gitmodules [hunk 4]
```diff
@@ -4,3 +4,6 @@
 [submodule "thirdparty/onnx"]
 	path = thirdparty/onnx
 	url = https://github.com/onnx/onnx.git
+[submodule "thirdparty/fp16"]
```
Comment (rdzhabarov) [inline .gitmodules]: i thought this is merged :)

Comment (qcolombet) [reply .gitmodules]: No yet. It will disappear when I rebase.

File: lib/Backends/Interpreter/InterpreterNodes.cpp (hunk 1 above)
Comment (qcolombet) [reply lib/Backends/Interpreter/InterpreterNodes.cpp]: Good point.
Done!

**Pull request merged**